src/punctuators.cpp

```diff
@@ -26,97 +26,97 @@ using namespace std;
-   { R"_(??')_", CT_CARET,        LANG_C | LANG_CPP | FLAG_DIG                                 }, // trigraph ^
-   { R"_(??<)_", CT_BRACE_OPEN,   LANG_C | LANG_CPP | FLAG_DIG                                 }, // trigraph {
-   { R"_(??>)_", CT_BRACE_CLOSE,  LANG_C | LANG_CPP | FLAG_DIG                                 }, // trigraph }
-   { R"_(??-)_", CT_INV,          LANG_C | LANG_CPP | FLAG_DIG                                 }, // trigraph ~
-   { R"_(??!)_", CT_ARITH,        LANG_C | LANG_CPP | FLAG_DIG                                 }, // trigraph |
+   { "!<=",      CT_COMPARE,      LANG_D                                                                 },
+   { "!<>",      CT_COMPARE,      LANG_D                                                                 },
+   { "!==",      CT_COMPARE,      LANG_D | LANG_ECMA                                                     },
+   { "!>=",      CT_COMPARE,      LANG_D                                                                 },
+   { "->*",      CT_MEMBER,       LANG_C | LANG_CPP | LANG_OC | LANG_D                                   },
+   { "...",      CT_ELLIPSIS,     LANG_C | LANG_CPP | LANG_OC | LANG_D | LANG_PAWN | LANG_JAVA | LANG_OC },
+   { "<<=",      CT_ASSIGN,       LANG_ALL                                                               },
+   { "<>=",      CT_COMPARE,      LANG_D                                                                 },
+   { "===",      CT_COMPARE,      LANG_D | LANG_ECMA                                                     },
+   { ">>=",      CT_ASSIGN,       LANG_ALL                                                               },
+   { ">>>",      CT_ARITH,        LANG_D | LANG_JAVA | LANG_PAWN | LANG_ECMA                             },
+   { "->*",      CT_MEMBER,       LANG_C | LANG_CPP | LANG_OC | LANG_D                                   },
+   { "%:@",      CT_POUND,        LANG_C | LANG_CPP | LANG_OC                                            }, // digraph  #@ MS extension
+   { R"_(??=)_", CT_POUND,        LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph #
+   { R"_(??()_", CT_SQUARE_OPEN,  LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph [
+   { R"_(??))_", CT_SQUARE_CLOSE, LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph ]
+   { R"_(??')_", CT_CARET,        LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph ^
+   { R"_(??<)_", CT_BRACE_OPEN,   LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph {
+   { R"_(??>)_", CT_BRACE_CLOSE,  LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph }
+   { R"_(??-)_", CT_INV,          LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph ~
+   { R"_(??!)_", CT_ARITH,        LANG_C | LANG_CPP | LANG_OC | FLAG_DIG                                 }, // trigraph |
```

Comment:
could've used the same language keywords inheritance concept here as well but these are fewer.

Reply:
True, I also noticed that I had added a duplicate entry for LANG_OC into CT_ELLIPSIS, fixed in a95b4e8b